source/extensions/upstreams/http/config.cc

```diff
@@ -202,7 +201,11 @@ ProtocolOptionsConfigImpl::ProtocolOptionsConfigImpl(
       header_validator_factory_(createHeaderValidatorFactory(options, server_context)),
       use_downstream_protocol_(options.has_use_downstream_protocol_config()),
       use_http2_(useHttp2(options)), use_http3_(useHttp3(options)),
-      use_alpn_(options.has_auto_config()) {}
+      use_alpn_(options.has_auto_config()) {
+  auto options_or_error = Http2::Utility::initializeAndValidateOptions(getHttp2Options(options));
```

Comment:
Can you move this code into a helper method, please, so that `http2_options_` constness is preserved? It is much easier to reason about thread safety of config objects if they are const correct.